Examples/Framework/src/EventData/SimMultiTrajectory.cpp

```diff
@@ -0,0 +1,49 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2019 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "ACTFW/EventData/SimMultiTrajectory.hpp"
+
+std::vector<FW::ParticleHitCount>
+FW::SimMultiTrajectory::identifyMajorityParticle(
+    const size_t& entryIndex) const {
+  std::vector<FW::ParticleHitCount> particleHitCount;
+  particleHitCount.reserve(10);
+  if (m_multiTrajectory) {
+    (*m_multiTrajectory).visitBackwards(entryIndex, [&](const auto& state) {
+      // No truth info with non-measurement state
+      if (not state.typeFlags().test(Acts::TrackStateFlag::MeasurementFlag)) {
+        return true;
+      }
+      // Find the truth particle associated with this state
+      const auto particleId = state.uncalibrated().truthHit().particleId();
```

Comment:
Theoretically this can fail. You're checking with the `MeasurementFlag` above, but is that always in sync? Also, I feel a bit uneasy thinking about this being a potential source of inconsistency. Essentially, the information whether a TS has a measurement or not is split between the TS Flag and the allocation itself now. Not sure what to do though.

Reply:
The `MeasurementFlag` is set along with creation of a measurement TS whose storage index for measurement is always valid. So the sync is guaranteed by the KF/CKF (at least for the moment). But you're right we could also do the check using the allocation directly.